qiskit/dagcircuit/dagcircuit.py

```diff
@@ -441,49 +444,40 @@ def compose_back(self, input_circuit, edge_map=None):
 
         Args:
             input_circuit (DAGCircuit): circuit to append
-            edge_map (dict): map {Bit: Bit} from the output wires of
-                input_circuit to input wires of self. The key and value
+            edge_map (dict): map {Bit: Bit} from the input wires of
+                input_circuit to output wires of self. The key and value
                 can either be of type Qubit or Clbit depending on the
                 type of the node.
 
         Raises:
             DAGCircuitError: if missing, duplicate or inconsistent wire
         """
-        edge_map = edge_map or {}
-
-        # Check the wire map for duplicate values
+        # if no edge_map given, try to do a 1-1 mapping in order
+        if edge_map is None:
+            if len(input_circuit.qubits()) > len(self.qubits()) or \
+               len(input_circuit.clbits()) > len(self.clbits()):
+                raise DAGCircuitError("Trying to compose with another DAGCircuit "
+                                      "which has more in edges.")
+            identity_qubit_map = dict(zip(input_circuit.qubits(), self.qubits()))
+            identity_clbit_map = dict(zip(input_circuit.clbits(), self.clbits()))
+            edge_map = {**identity_qubit_map, **identity_clbit_map}
+
+        # Check the edge_map for duplicate values
         if len(set(edge_map.values())) != len(edge_map):
             raise DAGCircuitError("duplicates in wire_map")
 
-        add_qregs = self._check_edgemap_registers(edge_map,
-                                                  input_circuit.qregs,
-                                                  self.qregs)
-        for qreg in add_qregs:
-            self.add_qreg(qreg)
-
-        add_cregs = self._check_edgemap_registers(edge_map,
-                                                  input_circuit.cregs,
-                                                  self.cregs)
-        for creg in add_cregs:
-            self.add_creg(creg)
-
-        self._check_wiremap_validity(edge_map, input_circuit.input_map,
-                                     self.output_map)
-
         # Compose
         for nd in input_circuit.topological_nodes():
             if nd.type == "in":
-                # if in wire_map, get new name, else use existing name
+                # if in edge_map, get new name, else use existing name
                 m_wire = edge_map.get(nd.wire, nd.wire)
```

Comment:
Shouldn't we use "reversed_edge_map" instead of `edge_map` here? (same goes for lines below, e.g. L. 485--489) Because we're going to change the definition of edge_map from output->input to input->output.

Reply:
Fixed at da3eeef79873cde85e0d45e79f06719a6b979d11 (edge_map keeps its definition)